examples/colormap_normalization/colormap_normalizations_piecewisenorm.py

```diff
@@ -0,0 +1,103 @@
+"""
+============================================
+Examples of arbitrary colormap normalization
+============================================
+
+Here I plot an image array with data spanning for a large dynamic range,
+using different normalizations. Look at how each of them enhances
+different features.
+
+"""
+
+
+import matplotlib.cm as cm
+import matplotlib.colors as colors
+import matplotlib.pyplot as plt
+
+from mpl_toolkits.mplot3d import Axes3D
+
+import numpy as np
+
+from sampledata import PiecewiseNormData
+
+X, Y, data = PiecewiseNormData()
+cmap = cm.spectral
+
+# Creating functions for plotting
+
+
+def make_plot(norm, label=''):
```

Comment:
> the user shouldn't need to know how you generated your data 'cause they'll have their own data anyway.

I agree that the user does not need to do this. The reason I made a separate file (sampledata.py) for generating the fake data instead of using cbook.get_sample_data is that sampledata.py is a 3KB file, while the equivalent npy file is 1 MB, and even if I reduce the resolution a bit , or the variable type, it will not get smaller than 100 KB. Do you know if there is any place to instead store cooked data, to store code that generates fake data?

> This doesn't have to be sensible in a real world way, it just has to show people what code will do what. You just don't want your fake data to become the example, 'cause it almost doesn't matter when showing norms.

Well I guess there is a manifold purpose for the examples:

1. Show the user how to use the normalization from a syntactic point of view. For this, a simple clear example with non sophisticated data will suffice.
2. Show the user what the normalization is actually doing to the data. For this having the line plot is really important, and most users will understand this better than nothing, because it visually shows without doubt the ranges used, and how the data will be compressed/mapped into the 0, 1 range.
3. Show the user what can be done with the normalization. And for this we need the examples with different kinds of small features to amplify them. I thing this is important also to understand what the normalization is for, because sometime people knows the problem they want to solve (i.e. amplify stuff in an array), but not what is the tool to solve it. 

If you really do not like the example showing this, I guess I would be happy to not use sophisticated cook data (number 2), as long as number 2 is covered. Otherwise by showing just the plots we are at risk of the user not following what is really going on.

Your thoughts?